src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,532 @@
+    delete this->dataPtr->imageBuffer;
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Init()
+{
+  return CameraSensor::Init();
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Load(sdf::ElementPtr _sdf)
+{
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(_sdf);
+  return this->Load(sdfSensor);
+}
+
+/////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Load(const sdf::Sensor &_sdf)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+  // BoundingBox Type
+  sdf::ElementPtr sdfElement = _sdf.Element();
+  if (sdfElement->HasElement("box_type"))
+  {
+    std::string type = sdfElement->Get<std::string>("box_type");
+
+    // convert type to lowercase
+    std::for_each(type.begin(), type.end(), [](char & c){
+      c = std::tolower(c);
+    });
+
+    if (type == "full_2d" || type == "full_box_2d")
+      this->dataPtr->type = BoundingBoxType::FullBox2D;
+    else if (type == "2d" || type == "visible_2d"
+      || type == "visible_box_2d")
+      this->dataPtr->type = BoundingBoxType::VisibleBox2D;
+    else if (type == "3d")
+      this->dataPtr->type = BoundingBoxType::Box3D;
+    else
+    {
+      ignerr << "Unknown bounding box type " << type << std::endl;
+      return false;
+    }
+  }
+
+  if (!Sensor::Load(_sdf))
+  {
+    return false;
+  }
+
+  // Check if this is the right type
+  if (_sdf.Type() != sdf::SensorType::BOUNDINGBOX_CAMERA)
+  {
+    ignerr << "Attempting to a load a BoundingBox Camera sensor, but received "
+      << "a " << _sdf.TypeStr() << std::endl;
+    return false;
+  }
+
+  if (_sdf.CameraSensor() == nullptr)
+  {
+    ignerr << "Attempting to a load a BoundingBox Camera sensor, but received "
+      << "a null sensor." << std::endl;
+    return false;
+  }
+
+  this->dataPtr->sdfSensor = _sdf;
+
+  this->dataPtr->topicBoundingBoxes = this->Topic();
+  this->dataPtr->topicImage = this->Topic() + "_image";
+
+  this->dataPtr->imagePublisher =
+    this->dataPtr->node.Advertise<ignition::msgs::Image>(
+      this->dataPtr->topicImage);
+
+  if (this->dataPtr->type == BoundingBoxType::Box3D)
+  {
+    this->dataPtr->boxesPublisher =
+      this->dataPtr->node.Advertise<
+      ignition::msgs::AnnotatedOriented3DBox_V>(
+      this->dataPtr->topicBoundingBoxes);
+  }
+  else
+  {
+    this->dataPtr->boxesPublisher =
+      this->dataPtr->node.Advertise<
+      ignition::msgs::AnnotatedAxisAligned2DBox_V>(
+      this->dataPtr->topicBoundingBoxes);
+  }
+
+  if (!this->dataPtr->imagePublisher)
+  {
+    ignerr << "Unable to create publisher on topic["
+      << this->dataPtr->topicImage << "].\n";
+    return false;
+  }
+
+  if (!this->dataPtr->boxesPublisher)
+  {
+    ignerr << "Unable to create publisher on topic["
+      << this->dataPtr->topicBoundingBoxes << "].\n";
+    return false;
+  }
+
+  if (!this->AdvertiseInfo())
+    return false;
+
+  if (this->Scene())
+  {
+    this->CreateCamera();
+  }
```

Comment:
If `this->Scene()` returns false (which means a scene doesn't exist), should we display an error message and return false?

Reply:
No it just will mark the initialized variable, then when the SetScene() is called will create the camera